src/components/queryExpressionEditor/buildFromResult/BuildFromResult.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import {ISearchBarStateProps, SearchBar} from '../../searchBar/SearchBar';
+import {QueryTrigger} from '../queryTrigger/QueryTrigger';
+import {IResult} from '../responseParser/ResponseParser';
+import {ResultList} from '../resultList/ResultList';
+
+export enum Field {
+    ObjectType = '@objecttype',
+    FileType = '@filetype',
+    ConnectorType = '@connectortype',
+    Source = '@source',
+}
+
+export interface IFieldExpression {
+    field: string;
+    fieldValue: string;
+}
+
+export interface IBuildFromResultProps {
+    queryTrigger: QueryTrigger;
+    updateQueryExpression: (expression: string) => void;
+}
+
+export interface IBuildFromResultState extends ISearchBarStateProps {
+    results: IResult[];
+}
+
+export class BuildFromResult extends React.Component<IBuildFromResultProps, IBuildFromResultState> {
+    constructor(props: IBuildFromResultProps) {
+        super(props);
+        this.state = {results: []};
+    }
+
+    async componentDidMount() {
+        const allResults = await this.props.queryTrigger.getAllResults();
+        this.setState({results: allResults});
+    }
+
+    private onClick(result: IResult) {
+        const parsedFieldExpression: string = this.getParsedFieldExpression(result);
+        this.props.updateQueryExpression(parsedFieldExpression);
+    }
+
+    private getParsedFieldExpression(result: IResult) {
+        const fieldExpression = this.getFieldExpression(result);
+        if (!fieldExpression) {
+            return null;
+        }
+        return this.buildFieldExpression(fieldExpression);
+    }
+
+    private getFieldExpression(result: IResult): IFieldExpression {
+        if (!result) {
+            return null;
+        }
+
+        if (result.objectType) {
+            return {field: Field.ObjectType, fieldValue: result.objectType};
+        }
+
+        if (result.fileType) {
+            return {field: Field.FileType, fieldValue: result.fileType};
+        }
+
+        if (result.connectorType) {
+            return {field: Field.ConnectorType, fieldValue: result.connectorType};
+        }
+
+        if (result.source) {
+            return {field: Field.Source, fieldValue: result.source};
+        }
+
+        return null;
+    }
+
+    private buildFieldExpression(fieldExpression: IFieldExpression): string {
+        if (fieldExpression.field && fieldExpression.fieldValue) {
+            return `${fieldExpression.field}=${fieldExpression.fieldValue}`;
+        } else {
+            return null;
+        }
+    }
+
+    private async onSearch(searchBarText: string) {

```

Comment:
IIRC `async` & `await` functions aren't supported by IE 11 which we need to support, use promises instead

Reply:
even promises will break unless `ES2018.Promise` is added in `lib` in the tsconfig. The quickest path to replace this would be to use jQuery which we try to move away from. I'm still not sure what is the best solution here, I think it will depend on whether we decide to keep this component in react-vapor or not also.

Reply:
in addition, the way we support asynchronous stuff for state management is usually through redux-thunk https://www.npmjs.com/package/redux-thunk , however that does not play well with react internal state , still continuing to think about possible options